src/result_management/read_results.py

```diff
@@ -64,3 +65,212 @@ def extract_dataset_from_h5(dataset: h5py.Dataset) -> list:
+                                        if output_name not in output_dict[case]:
+                                            output_dict[case][output_name] = tec_output
+                                    except KeyError:
+                                        pass
+
+                if "Networks" in component_set:
+                    df = extract_datasets_from_h5group(
+                        hdf_file["design/networks"]
+                    ).sum()
+                    for period in df.index.levels[0]:
+                        for netw in df.index.levels[1]:
+                            parameters1 = [
+                                "para_capex_gamma1",
+                                "para_capex_gamma2",
+                                "para_capex_gamma3",
+                                "para_capex_gamma4",
+                            ]
+                            for para in parameters1:
+                                output_name = f"{period}/{netw}/{para}"
+                                try:
+                                    netw_output = df.loc[period, netw, para]
+                                    if case not in output_dict:
+                                        output_dict[case] = {}
+                                    if output_name not in output_dict[case]:
+                                        output_dict[case][output_name] = (
+                                            netw_output.iloc[0]
+                                        )
+                                except KeyError:
+                                    pass
+                            for arc in df.index.levels[2]:
+                                parameters2 = ["size", "capex"]
+                                for para in parameters2:
+                                    output_name = f"{period}/{netw}/{arc}/{para}"
+                                    try:
+                                        arc_output = df.loc[period, netw, arc, para]
+                                        if case not in output_dict:
+                                            output_dict[case] = {}
+                                        if output_name not in output_dict[case]:
+                                            output_dict[case][output_name] = arc_output
+                                    except KeyError:
+                                        pass
+
+                if "Import" in component_set:
+                    df = extract_datasets_from_h5group(
+                        hdf_file["operation/energy_balance"]
+                    )
+                    for period in df.columns.levels[0]:
+                        for node in df.columns.levels[1]:
+                            for car in df.columns.levels[2]:
+                                parameters = ["import", "import_price"]
+                                for para in parameters:
+                                    try:
```

Comment:
see above. Here it is even more 'dangerous', because we have a lot of statments that might produce an error, and we just skip them regardless of the error we get.

Reply:
done